Datasnap/Classes/Utilities/Datasnap.m

```diff
@@ -59,21 +68,12 @@ - (id)initWithApiKey:(NSString*)apiKey
     [self initializeData];
     return self;
 }
-+ (id)sharedClient
-{
-    static Datasnap* sharedClient = nil;
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        sharedClient = [self new];
-    });
-    return sharedClient;
-}
+
 - (void)initializeData
 {
-    if (self.googleAdOptIn) {
-        googleAd = @"YES";
-        self.mobileDeviceIosIdfa = [self identifierForAdvertising];
-    }
+    self.device = [[Device alloc] init];
+    NSString *adOptedIn = self.googleAdOptIn ? @"YES" : @"NO";
```

Comment:
@almcintyre, @DevinJohnson : I don't like variables that are only used once and set with simple logic. Move to place of use.